private/bufpkg/bufprotoplugin/response_writer.go

```diff
@@ -110,48 +111,43 @@ func writeInsertionPoint(
 	// bufio.Scanner.Scan() inline
 	newline := []byte{'\n'}
 	var found bool
-	for targetScanner.Scan() {
+	for i := 0; targetScanner.Scan(); i++ {
+		if i > 0 {
+			// These writes cannot fail, they will panic if they cannot allocate.
+			_, _ = postInsertionContent.Write(newline)
+		}
+
 		targetLine := targetScanner.Bytes()
 		if !bytes.Contains(targetLine, match) {
-			// these writes cannot fail, they will panic if they cannot
-			// allocate
+			// These writes cannot fail, they will panic if they cannot allocate.
 			_, _ = postInsertionContent.Write(targetLine)
-			_, _ = postInsertionContent.Write(newline)
 			continue
 		}
 		// For each line in then new content, apply the
 		// same amount of whitespace. This is important
 		// for specific languages, e.g. Python.
 		whitespace := leadingWhitespace(targetLine)
 
-		// Create another scanner so that we can seamlessly handle
-		// newlines in a platform-agnostic manner.
-		insertedContentScanner := bufio.NewScanner(bytes.NewBufferString(insertionPointFile.GetContent()))
-		insertedContent := scanWithPrefixAndLineEnding(insertedContentScanner, whitespace, newline)
-		// This write cannot fail, it will panic if it cannot
-		// allocate
-		_, _ = postInsertionContent.Write(insertedContent)
+		// Insert the content from the insertion point file. Handle newlines in
+		// a platform-agnostic manner.
+		insertedContentReader := strings.NewReader(insertionPointFile.GetContent())
+		writeWithPrefixAndLineEnding(postInsertionContent, insertedContentReader, whitespace, newline)
```

Comment:
Write to the buffer instead of allocating a new slice and then writing to the buffer.